sdk/dotnet/Pulumi/Core/InputList.cs

```diff
@@ -147,7 +147,8 @@ IEnumerator IEnumerable.GetEnumerator()
 
         public async IAsyncEnumerator<Input<T>> GetAsyncEnumerator(CancellationToken cancellationToken)
         {
-            var data = await _outputValue.GetValueAsync().ConfigureAwait(false);
```

Comment:
This changes behavior, but is reasonable? If someone uses an unknown list as an GetAsyncEnumerator, this will treat it as empty. Before, it would have thrown an exception. We have options here: we can restore previous behavior with `whenUnknown: default!` or we can cook up a more descriptive exception.